tests/modules/ctc/ctc_backend_tests.cc

```diff
@@ -141,21 +168,48 @@ TEST(CtcBackEndTest, TrainReceiveBlockOccupancy)
     block_states.push_back(block_state_1);
     ctc.SetBlockStates(types::TrackId::TRACKID_GREEN, block_states);
 
-    train1 = ctc.GetTrainById(1);
+    ctc.GetTrainById(1, train1);
     ASSERT_EQ(train1.authority.front(), 64);
-    ASSERT_EQ(train1.current_position, 63);
-    ASSERT_EQ(train1.suggested_speed, 19);
+    ASSERT_EQ(ctc.GetTrainCurrentPosition(1), 63);
+    ASSERT_EQ(ctc.GetTrainSuggestedSpeed(1), 19);
+}
+
+TEST(CtcBackEndTest, ManualDispatchToMultipleBlock)
+{
+    ctc::Ctc ctc(types::TrackId::TRACKID_GREEN);
+    ctc.ManualDispatch(1, 67);
+    ctc.ManualDispatch(1, 70);
+    ctc::Train train1;
+    ctc.GetTrainById(1, train1);
+    ASSERT_EQ(train1.current_position, 0);      // Sitting in yard
+    ASSERT_EQ(train1.authority.size(), 5);      // Authority = 5 blocks to get to block 67
+    ASSERT_EQ(train1.destination_list[CTC_TRAIN_CURRENT_DESTINATION].destination, 67);
+    ASSERT_EQ(train1.destination_list[1].destination, 70);
+
+    ctc.SetBlockStates(types::TrackId::TRACKID_GREEN, { types::BlockState(63, true, false) });
+    ctc.SetBlockStates(types::TrackId::TRACKID_GREEN, { types::BlockState(64, true, false) });
+    ctc.SetBlockStates(types::TrackId::TRACKID_GREEN, { types::BlockState(65, true, false) });
+    ctc.SetBlockStates(types::TrackId::TRACKID_GREEN, { types::BlockState(66, true, false) });
+    ctc.SetBlockStates(types::TrackId::TRACKID_GREEN, { types::BlockState(67, true, false) });
+    ctc.GetTrainById(1, train1);
+    ASSERT_EQ(train1.current_position, 67);
+    ASSERT_EQ(train1.authority.size(), 3);
+    ASSERT_EQ(train1.destination_list[CTC_TRAIN_CURRENT_DESTINATION].destination, 70);
+
+    ctc.SetBlockStates(types::TrackId::TRACKID_GREEN, { types::BlockState(68, true, false) });
+    ctc.SetBlockStates(types::TrackId::TRACKID_GREEN, { types::BlockState(69, true, false) });
+    ctc.SetBlockStates(types::TrackId::TRACKID_GREEN, { types::BlockState(70, true, false) });
+    ctc.GetTrainById(1, train1);
+    ASSERT_EQ(train1.current_position, 70);
+    ASSERT_EQ(train1.authority.size(), 163);
+    ASSERT_EQ(train1.destination_list[CTC_TRAIN_CURRENT_DESTINATION].destination, 0);
 }
 
 TEST(CtcBackEndTest, GetSuggestedSpeedAndAuthorities)
```

Comment:
This test should check that the track is correctly set in the data that `GetSuggestedSpeedAndAuthority` returns so you are testing for the bug and verify the bugfix works.

Reply:
Good call. Done!